topics/client-engines.md

```diff
@@ -60,181 +67,236 @@ The table below shows whether a specific engine supports HTTP/2 and [WebSockets]
+In addition to the [`ktor-client-core`](client-dependencies.md) artifact, the Ktor client requires a dependency for a
+specific engine. Each supported platform has a set of available engines, described in the corresponding sections:
+
 * [JVM](#jvm)
 * [JVM and Android](#jvm-android)
 * [JavaScript](#js)
 * [Native](#native)
 
-> For different engines, Ktor provides platform-specific artifacts with suffixes such as `-jvm` or `-js`, for example, `ktor-client-cio-jvm`. Gradle resolves artifacts appropriate for a given platform while Maven doesn't support this capability. This means that for Maven, you need to add a platform-specific suffix manually.
+> Ktor provides platform-specific artifacts with suffixes such as `-jvm` or `-js`. For example, `ktor-client-cio-jvm`.
+> Dependency resolution differs by build tool. While Gradle resolves artifacts appropriate for a given platform, Maven
+> doesn't support this capability. This means that for Maven, you need to specify the platform suffix manually.
 >
 {type="note"}
 
-## Create a client with a specified engine {id="create"}
-To create the HTTP client with a specific engine, pass an engine class as an argument to the [HttpClient](https://api.ktor.io/ktor-client/ktor-client-core/io.ktor.client/-http-client/index.html) constructor. For example, you can create a client with the `CIO` engine as follows:
+## Specify an engine {id="create"}
+
+To use a specific engine, pass the engine class to the [
+`HttpClient`](https://api.ktor.io/ktor-client/ktor-client-core/io.ktor.client/-http-client/index.html) constructor. The
+following example creates a client with the `CIO` engine:
+
 ```kotlin
 ```
+
 {src="snippets/_misc_client/CioCreate.kt"}
 
-### Default engine {id="default"}
-If you call the `HttpClient` constructor without an argument, the client will choose an engine automatically depending on the artifacts [added in a build script](#dependencies).
+## Default engine {id="default"}
+
+If you omit the engine argument, the client will choose an engine automatically based on the dependencies
+[in your build script](#dependencies).
+
 ```kotlin
 ```
-{src="snippets/_misc_client/DefaultEngineCreate.kt"}
 
-This can be useful for multiplatform projects. For example, for a project targeting both [Android and iOS](client-create-multiplatform-application.md), you can add the [Android](#jvm-android) dependency to the `androidMain` source set and the [Darwin](#darwin) dependency to the `iosMain` source set. The necessary dependency will be selected at compile time.
+{src="snippets/_misc_client/DefaultEngineCreate.kt"}
 
+This is especially useful in multiplatform projects. For example, for a project targeting
+both [Android and iOS](client-create-multiplatform-application.md), you can add the [Android](#jvm-android) dependency
+to the `androidMain` source set and the [Darwin](#darwin) dependency to the `iosMain` source set. The appropriate
+engine is selected at compile time.
```

Comment:
Oh, this wasn't true. The engine is selected at run time, at the moment of `HttpClient` creation